### PR TITLE
Use invisible POST form for access key redirect

### LINKED ISSUE #77: In MaybeRedirect when redirect to login, use wp_redirect and exit.
safe redirect doesn't work for cross-origin redirect. Have to exit.

### PR DESCRIPTION
**Pull request by Shelob9**
Fixes #77

### CODE REVIEW

**Comment by Shelob9** [comment]
@zackkatz I kept the form, and it submits the same way. BUT, if there is a match, it prints out a form (based on this https://github.com/trustedlogin/trustedlogin-vendor/blob/cb61fdec95552e58a7d572e17e9ecf88cc274657/assets/trustedlogin-access-keys.js#L21-L31) and submits that forms. Seams to work with my test.

File: admin/components/AccessKeyForm.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ const AccessKeyForm = ({ initialAccountId = null }) => {
     if (accessKey) {
       action = `${action}&ak=${accessKey}`;
     }
+    console.log({ action });
```
Comment (zackkatz) [inline admin/components/AccessKeyForm.js:42]: To be removed

File: admin/components/Sections.js [hunk 2]
```diff
@@ -75,7 +79,7 @@ export const DangerZone = () => {
           <TitleDescriptionLink title={__("Are You Sure?")} />
           <SubmitAndCanelButtons
             onSubmit={onDelete}
-            submitText={"Delete Team"}
+            submitText={"Rest Keys"}
```
Comment (zackkatz) [inline admin/components/Sections.js:82]: ~Rest~ Reset

File: php/MaybeRedirect.php [hunk 3]
```diff
@@ -82,11 +82,43 @@ public static function handle()
...
 			}
-
 			$handler = new AccessKeyLogin();
-			$parts_or_error = $handler->handle();
-			if( is_array($parts_or_error)){
-				wp_redirect( $parts_or_error['loginurl'] );
+			$parts = $handler->handle();
+			$parts = \json_decode('
+			{
+				"identifier": "F6qvCYIze7i/Ekx4GaYYpNwrlepY8IABPsJFihZD3FRN57eBzx3U3T3HdsRu6cfe9Rr94eihV68dFj3++BNwK7xqrDYJA6Ha+r7vYKwx5vf9tf+/vaLwrro0mLE+9SXcn1cMeFbNQ5v/gz86gu7Qz23QkOsy7mFC2ffU4+dIR8IuOPrdIxGWWs+tPBkaXUn2",
```
Comment (zackkatz) [inline php/MaybeRedirect.php:89]: Just confirmation that this is placeholder 😉

File: php/MaybeRedirect.php [hunk 4]
```diff
@@ -82,11 +82,43 @@ public static function handle()
...
+					</form>
+				</div>',
+					esc_url($parts['loginurl']),
+					esc_attr($parts['endpoint']),
+					esc_attr($parts['identifier'])
+
+				);
+				$output .= ('<script>
+					window.addEventListener("load", () => {
+						document.getElementById("trusted-login-redirect-form").submit();
```
Comment (zackkatz) [inline php/MaybeRedirect.php:118]: ~`trusted-login`~ `trustedlogin`

File: php/MaybeRedirect.php [hunk 5]
```diff
@@ -82,11 +82,43 @@ public static function handle()
...
+				"siteurl": "https://trustedlogin.support"
+			  }', true );
+
+			$parts['endpoint'] = \trustedlogin_vendor()->getEncryption()::hash($parts['siteurl'] . $parts['identifier']);
+			$parts['loginurl'] = $parts['siteurl'] . '/' . $parts['endpoint'] . '/' . $parts['identifier'];
+			if( is_array($parts)){
+				$output = sprintf('<div>
+					<form
+						method="post"
+						id="trusted-login-redirect-form"
```
Comment (zackkatz) [inline php/MaybeRedirect.php:102]: ~`trusted-login`~ `trustedlogin`

**Review by zackkatz — CHANGES_REQUESTED** [review]

File: admin/components/Sections.js [hunk 6]
```diff
@@ -64,7 +66,9 @@ export const DangerZone = () => {
                   </p>
                 </div>
               </div>
-              <DangerButton onClick={onClick} id="sort-menu-button">
+              <DangerButton
+                onClick={() => setIsResetting(true)}
+                id="rest-keys-button">
```
Comment (zackkatz) [inline admin/components/Sections.js:71]: ~`rest-keys-button`~ `reset-keys-button`

**Comment by Shelob9** [comment]
@zackkatz I tested this ZIP file. Ther request to get the access key returned an expected error for invalid secrets (my test site is bad), and it returned expected error.  Can you please retest this, with Gravity View https://github.com/trustedlogin/vendor/blob/af4ea5ad0cb6c47342a1c794900d51fdb4b1476a/trustedlogin-vendor.zip

I'm going to eat lunch and then add better handling of the errors, which are now being caught properly.

**Comment by Shelob9** [comment]
@zackkatz [4dc2592](https://github.com/trustedlogin/vendor/pull/82/commits/4dc2592d03ec068c89d85c0bb12268cae8a89c27) impliments the fetch to `loginurl` and then `window.location` redirect to `siteurl`. 

In 4dc2592d03ec068c89d85c0bb12268cae8a89c27 I told fetch to accept cookies set during cross-origin request. That should help. Not sure.

**Comment by Shelob9** [comment]
@zackkatz I think this can be merged.

**Pull request merged**